webrtc/RTCConfiguration-iceCandidatePoolSize.html

```diff
@@ -1,35 +1,109 @@
-{
-    test(function() {
-        var pc = new RTCPeerConnection({iceCandidatePoolSize: size})
-        assert_equals(pc.getConfiguration().iceCandidatePoolSize, expectedSize !== undefined ? expectedSize : size)
-    }, "Setting iceCandidatePoolSize to a valid value: " + size)
-}
-
-function testInvalidPoolSize(size)
-{
-    test(function() {
-        assert_throws(new TypeError(), () => new RTCPeerConnection({iceCandidatePoolSize: size}))
-    }, "Setting iceCandidatePoolSize to an invalid value: " + size)
-}
-
-testValidPoolSize(10)
-testValidPoolSize(0)
-testValidPoolSize(null, 0)
-testValidPoolSize(undefined, 0)
-
-testInvalidPoolSize(-1)
-testInvalidPoolSize(1000)
-testInvalidPoolSize("string")
+
+/*
+
+dictionary RTCConfiguration {
+    ...
+    [EnforceRange]
+    octet                    iceCandidatePoolSize = 0;
+};
+
+... of type octet
+*/
+test(() => {
+  const pc = new RTCPeerConnection();
+  assert_equals(pc.getConfiguration().iceCandidatePoolSize, 0);
+}, "Initialize a new RTCPeerConnection with no iceCandidatePoolSize");
+
+test(() => {
+  const pc = new RTCPeerConnection({
+    iceCandidatePoolSize: 0
+  });
+  assert_equals(pc.getConfiguration().iceCandidatePoolSize, 0);
+}, "Initialize a new RTCPeerConnection with iceCandidatePoolSize: 0");
+
+test(() => {
+  const pc = new RTCPeerConnection({
+    iceCandidatePoolSize: 255
+  });
+  assert_equals(pc.getConfiguration().iceCandidatePoolSize, 255);
+}, "Initialize a new RTCPeerConnection with iceCandidatePoolSize: 255");
+
+test(() => {
+  assert_throws(new TypeError(), () => {
+    new RTCPeerConnection({
+      iceCandidatePoolSize: -1
+    });
+  });
+}, "Initialize a new RTCPeerConnection with iceCandidatePoolSize: -1 (Out Of Range)");
+
+test(() => {
+  assert_throws(new TypeError(), () => {
+    new RTCPeerConnection({
+      iceCandidatePoolSize: 256
+    });
+  });
+}, "Initialize a new RTCPeerConnection with iceCandidatePoolSize: 256 (Out Of Range)");
+
+
+/*
+Reconfiguration
+*/
+const pc = new RTCPeerConnection({});
+
+test(() => {
+  pc.setConfiguration({
+    iceCandidatePoolSize: 0
+  });
+  assert_equals(pc.getConfiguration().iceCandidatePoolSize, 0);
+}, "Reconfigure RTCPeerConnection instance iceCandidatePoolSize to 0");
+
+test(() => {
+  pc.setConfiguration({
+    iceCandidatePoolSize: 255
+  });
+  assert_equals(pc.getConfiguration().iceCandidatePoolSize, 255);
+}, "Reconfigure RTCPeerConnection instance iceCandidatePoolSize to 255");
+
+/*
+The following tests include an explicit assertion for the existence of a
+setConfiguration function to prevent the assert_throws from catching the
+TypeError object that will be thrown when attempting to call the
+non-existent setConfiguration method (in cases where it has not yet
+been implemented). Without this check, these tests will pass incorrectly.
+*/
+
+test(() => {
+  assert_equals(typeof pc.setConfiguration, "function", "RTCPeerConnection.prototype.setConfiguration is not implemented");
```

Comment:
Nit: It is probably more clear if we merge this test with the previous test, since the first success case would have throw and abort the remaining assertions. But I get why having the tests done separately is clearer in most other cases, so either way is fine.